docs/src/main/asciidoc/security-authentication-mechanisms.adoc

```diff
@@ -93,6 +93,53 @@ quarkus.http.auth.form.login-page=
 quarkus.http.auth.form.error-page=
 ----
 
+Now that you have disabled redirects for the SPA, you must login and logout programmatically from your client.
+Below are example JavaScript methods for logging into the `j_security_check` endpoint and logging out of the application by destroying the cookie.
+
+[source,javascript]
+----
+const login = () => {
+    // Create an object to represent the form data
+    const formData = new URLSearchParams();
+    formData.append("j_username", username);
+    formData.append("j_password", password);
+
+    // Make an HTTP POST request using fetch against j_security_check endpoint
+    fetch("j_security_check", {
+        method: "POST",
+        body: formData,
+        headers: {
+            "Content-Type": "application/x-www-form-urlencoded",
+        },
+    })
+    .then((response) => {
+        if (response.status === 200) {
+            // Authentication was successful
+            console.log("Authentication successful");
+        } else {
+            // Authentication failed
+            console.error("Invalid credentials");
+        }
+    })
+    .catch((error) => {
+        console.error(error);
+    });
+};
+----
+
+To logout of the SPA you must destroy the cookie and redirect back to your main page.
+
+[source,javascript]
+----
+const logout= () => {
+    // delete the credential cookie essentially killing the session
+    const removeCookie = `quarkus-credential=; Max-Age=0;path=/`;
```

Comment:
Does it work out of the box, with the httpOnly flag set to true by default ?

Reply:
@sberyozkin out of the box it looks like httpOnly is set to `false` by default:
![image](https://github.com/quarkusio/quarkus/assets/4399574/2ef1f7ef-8aad-428f-80e6-33b936ed4b59)

Reply:
@melloware thanks, https://github.com/quarkusio/quarkus/blob/main/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/FormAuthConfig.java#L127 indeed have a default false value. It should be set to true by default, but it is another issue. 

So IMHO it has to be noted that this solution only works with the `HttpOnly` disabled, and I wonder if it is a good idea to recommend a logout solution which depends on the cookie being accessible to JavaScript.

Should a logout handler solution be recommended ?

Reply:
Good points.  I guess I should mention that it has to be httponly false and also provide an example server side controller logout?

Reply:
>  It should be set to true by default

+1

>  I wonder if it is a good idea to recommend a logout solution which depends on the cookie being accessible to JavaScript.

Here @melloware tries to document logout for SPA. SPA FWs actively prevents XSS unless you explicitly override it, in which is you are aware of what you are doing. All points you raised are absolutely true, but what is recommended here is not dangerous for typical SPA.

That is why I suggested this new code examples should be rather treated as meta code, because JS + AJAX doesn't provide OOTB protection nor the typical way you handle things (how you store state, route, ...) with SPA.

> Good points. I guess I should mention that it has to be httponly false and also provide an example server side controller logout?

+1

I'd also mention why - XSS and make it clear when it is needed.

Reply:
This thread is on `logout`, same is true for `login` documented here.